src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -5213,6 +5213,19 @@ int RGWRados::restore_obj_from_cloud(RGWLCCloudTierCtx& tier_ctx,
                       return 0;
                     });
 
+  // fetch mtime of the object and other attrs of the object
+  // to check for restore_status
+  RGWRados::Object op_target(this, dest_bucket_info, obj_ctx, dest_obj);
+  RGWRados::Object::Read read_op(&op_target);
+  read_op.params.lastmod = &mtime;
```

Comment:
are we reading the mtime in ` RadosObject::restore_obj_from_cloud` and passed to this function earlier in LN2570.